src/mqt/ddsim/qasmsimulator.py

```diff
@@ -73,22 +74,57 @@ def target(self):
     def max_circuits(self):
         return None
 
-    def run(self, quantum_circuits: QuantumCircuit | list[QuantumCircuit], **options: dict[str, Any]) -> DDSIMJob:
+    @staticmethod
+    def _bind_parameters(
+        quantum_circuits: list[QuantumCircuit], parameter_values: Sequence[Sequence[float]] | None = None
```

Comment:
According to Qiskit, the proper type to pass to `assign_parameters` or `bind_parameters` is "[Mapping](https://docs.python.org/3/library/typing.html#typing.Mapping)[[Parameter](https://qiskit.org/documentation/stubs/qiskit.circuit.Parameter.html#qiskit.circuit.Parameter), [float](https://docs.python.org/3/library/functions.html#float)] | [Sequence](https://docs.python.org/3/library/typing.html#typing.Sequence)[[float](https://docs.python.org/3/library/functions.html#float)]"

See https://qiskit.org/documentation/stubs/qiskit.circuit.QuantumCircuit.html#qiskit.circuit.QuantumCircuit.bind_parameters
Please adapt accordingly throughout this PR (including a test for both versions.

Reply:
Done :)